src/chapters/control-flow/loops.rst

```diff
@@ -0,0 +1,278 @@
+Loops
+=====
+
+.. index:: ! for loop
+
+``for`` Loop
+------------
+
+In Java we write a definite loop (aka a **for loop**) as:
+
+.. sourcecode:: java
+   :linenos:
+
+   for (int i = 0; i < 10; i++ ) {
+      System.out.println(i);
+   }
+
+.. note:: 
+
+   You may not be familiar with the expression ``i++`` since it is not
+   found in all languages. The ``++`` is an increment operator that has the same
+   effect as ``i += 1``. In this example, since the ``++`` comes after
+   ``i``, we call it a postfix increment operator. There is also a ``--``
+   decrement operator in Java. For more information, see 
+   `Increment and Decrement Operators <http://www.javawithus.com/tutorial/increment-and-decrement-operators>`__.
+
+
+The Java ``for`` loop gives you explicit control over the starting, stopping, and stepping 
+of the loop variable inside the parentheses. You can think of it this way:
+
+.. sourcecode:: java
+   :linenos:
+
+   for (start clause; stop clause; step clause) {
+      statement1
+      statement2
+      ...
+   }
+
+If you want to start at 100, stop at 0 and count backward by 5, the loop is written as:
+
+.. sourcecode:: java
+   :linenos:
+
+   for (int i = 100; i >= 0; i -= 5) {
+      System.out.println(i);
+   }
+
+.. index:: ! for-each loop
+
+``for-each`` Loop
+-------------
+
+Java also provides a syntax to iterate over any sequence or collection, such as an Array: 
+
+.. sourcecode:: java
+   :linenos:
+
+   int k[] = {1, 1, 2, 3, 5, 8, 13, 21};
+
+   for (int i : k) {
+      System.out.println(i);
+   }
+
+Here, the loop variable moves through the items in the Array of integers, ``k[]``. The syntax
+here uses a colon symbol, ``:``. This type of loop is known as a **for-each loop**.
+
+.. tip::
+
+   When considering this structure, it can be helpful to read the code sample above to yourself
+   as "For each integer in ``Array k``...".
+
+This loop version also works with a String, where we
```

Comment:
Code highlighting for ``String`` and ``Array``.

Reply:
i think syntax highlighting each use of these terms gets distracting. I think it's helpful to distinguish between capitalized and not capitalized variable types but giving each reference to a variable type syntax highlighting is pretty cluttering imo

Reply:
Yeah, I can see that.  But with String as a data type, this gets kinda cloudy.
Maybe:
1) ``String variableName``,
2) "...of type ``String``", but not for '"xyz" is a string' (lowercase).
3) ``Capitalized`` vs. lowercase when we want syntax highlighting?

Reply:
1. makes sense when we're using the variable name but there are plenty of instances where we won't want to do that and refer to what can be done on a String or Array, etc in general.
2. I do think it's important to keep consistent with using the uppercase name. it will get more confusing in the int/Int, double/Double, etc territory. 
3. I think this option can set us up for inconsistencies.
4. We do decide to syntax highlight every reference to a type name. this is what the oracle docs do. however, they dont use a different color for the code style, which i think helps visually.

Let's table this for now and bring it up in the working meeting tomorrow. I'll add these options to the doc to discuss.

Reply:
OK.